expected/wasm32-wasi-threads/predefined-macros.txt

```diff
@@ -1499,6 +1499,15 @@
 #define RRFIXEDSZ NS_RRFIXEDSZ
 #define RRQ 01
 #define RS_HIPRI 0x01
+#define RTLD_DEFAULT ((void *)0)
+#define RTLD_DI_LINKMAP 2
+#define RTLD_GLOBAL 256
+#define RTLD_LAZY 1
+#define RTLD_LOCAL 0
```

Comment:
isn't it straightforward to use a non-zero value?
an implementation can still choose to ignore it.

Reply:
@dicej ping

Reply:
Sorry, I didn't notice these comments earlier.  The values in these files come straight from libc-top-half/musl/include/dlfcn.h, which appears to remain unchanged since it was imported from musl libc.  `glibc` seems to use the same values for these constants also.  I'm not necessarily opposed to changing the values, but sticking with the musl values seems like a safe default here.

Reply:
Today I learned that `RTLD_GLOBAL` is the default on MacOS, but `RTLD_LOCAL` is the default on Linux (which I guess is why `RTLD_LOCAL` is zero on musl and glibc but nonzero on MacOS).  I guess we ought to decide that the default should be for WASI.

Reply:
unless you want to decide the default right now, we can postpone the decision by making it a non-zero value.

Reply:
I just pushed an update changing it to a non-zero value.

Reply:
I'm not sure I understand how we can make both of these non-zero.   This is a binary choice between global or local symbol binding, its not possible to choose neither.   The question is what the default should be when neither is specified and the answer has to be one or the other.

I suggest we go with that musl's defaults for now given that we are largely a port of musl here.

Reply:
Oh wait, I understand now.  Please ignore my last comment.  Having `RTLD_LOCAL` and `RTLD_GLOBAL` both be non-zero allows the implementation to detect the default case and choose one or the other at runtime.     SGTM